fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt

```diff
@@ -115,7 +125,20 @@ class TranslationsDialogFragment : BottomSheetDialogFragment() {
                                             from = BrowserDirection.FromTranslationsDialogFragment,
                                         )
                                     },
-                                    onTranslateButtonClick = {},
+                                    onTranslateButtonClick = {
+                                        store.getDetectedLanguages()?.let {
```

Comment:
I don't think it is okay to call methods from the store directly from fragments.You should call in fragment a method from interactor, from interector to controller, and then in controller the store.

Reply:
Thanks for mentioning this, was wondering about that!

Reply:
Updated, so that the `translate` will revert to some sensible use cases for nulls to get this out of the fragment.